hearts/game/manager.py

```diff
@@ -102,9 +118,10 @@ def play_trick(self) -> None:
             key=lambda x: x.value if x.value > 1 else 99,
```

Comment:
What is this line doing? What is x.value here?

Reply:
This is part of sorting in python. If you gave the `sort` function a list of integers, it would sort them ascending like this:
```python
my_list = [5, 3, 4, 6, 1, 2]
my_list.sort()
# my_list now looks like this: [1, 2, 3, 5, 6]
```
if you give it strings it will look like this:
```python
my_list = ['c', 'd', 'a', 'b']
my_list.sort()
# my_list now looks like this: ['a', 'b', 'c', 'd']
```

This is great for simple data types but what if you try to sort a list of dictionaries?
```python
my_list = [{'some_key': 2}, {'some_key': 3}, {'some_key': 1}]
my_list.sort()
????
```

This is where the `key` parameter comes in handy. You can define a function that modifies the value you want to sort. Each element of the list will be fed through the function then sorted.
```python
def sort_on_some_key(list_element: dict) -> int:
    return list_element['some_key']

my_list = [{'some_key': 2}, {'some_key': 3}, {'some_key': 1}]
my_list.sort(key=sort_on_some_key)
# my_list now looks like this: [{'some_key': 1}, {'some_key': 2}, {'some_key': 3}]
```

So in the actual example you highlighted, I am sorting a list of `Card` objects. These objects have a `value` attribute that I want to sort on.
```python
def sort_on_value(list_element: dict) -> int:
    return list_element.value

my_list = [Card(value=3), Card(value=1)]
my_list.sort(key=sort_on_value)
# my_list now looks like this: [Card(value=1), Card(value=3)]
```

The only thing tricky here is an Ace has a value of 1 but I want to sort it as higher than a King which has a value of 12. To do this I am using a [ternary](https://book.pythontips.com/en/latest/ternary_operators.html) expression which is equivalent to this:
```python
if x.value > 1
    return x.value
else:
    return 99
```

I recommend skimming these docs to get an idea of how it works:
https://docs.python.org/3/howto/sorting.html